packages/core/src/integrations/mcp-server/attributeExtraction.ts

```diff
@@ -45,21 +45,27 @@ import type {
  * @returns Transport type mapping for span attributes
  */
 export function getTransportTypes(transport: MCPTransport): { mcpTransport: string; networkTransport: string } {
-  const transportName = transport.constructor?.name?.toLowerCase() || '';
-
-  if (transportName.includes('stdio')) {
-    return { mcpTransport: 'stdio', networkTransport: 'pipe' };
+  if (!transport?.constructor) {
+    return { mcpTransport: 'unknown', networkTransport: 'unknown' };
   }
+  const transportName = transport.constructor.name?.toLowerCase() || 'unknown';
 
-  if (transportName.includes('streamablehttp') || transportName.includes('streamable')) {
-    return { mcpTransport: 'http', networkTransport: 'tcp' };
+  if (transportName === 'unknown') {
```

Comment:
```suggestion
  const transportName = transport.constructor.name?.toLowerCase();

  if (!transportName) {
```

should IMHO be clear enough, or can the constructor name actually be `unknown` and we want it to behave that way?

Reply:
You're right! thanks for reviewing this. it couldn't be `unknown`. fixed!